examples/c/src/phi3.cpp

```diff
@@ -97,31 +142,22 @@ void CXX_API(const char* model_path) {
 
     auto generator = OgaGenerator::Create(*model, *params);
 
-    while (!generator->IsDone()) {
-      generator->ComputeLogits();
-      generator->GenerateNextToken();
+    std::thread th(Generate_Output_CXX, generator.get(), std::move(tokenizer_stream), is_first_token, std::ref(timing));
```

Comment:
Can you reverse this? Make the main thread do the generation, while the newly created thread checks for the interrupt? That will reduce the changes in this example and users who do not care about termination, can still look at the main function code.

Reply:
Thanks, this is a good idea.